ref(theme): Use more readable accent colors in dark mode

vuluongj20 opened the pull request:
2 things:
* Update accent colors in dark mode for better readability
* Update color aliases: use ___400 accent colors for text

**Before:**
<img width="825" alt="Screen Shot 2022-11-15 at 11 38 47 AM" src="https://user-images.githubusercontent.com/44172267/202010756-7622b0d4-ca60-45aa-878c-39251209227b.png">
<img width="245" alt="Screen Shot 2022-11-15 at 11 39 32 AM" src="https://user-images.githubusercontent.com/44172267/202010753-35dbd439-ba14-4ac5-9d93-1a41fb35e048.png">


**After:**
<img width="825" alt="Screen Shot 2022-11-15 at 11 38 59 AM" src="https://user-images.githubusercontent.com/44172267/202010754-5c6d6e93-6c49-46c8-a916-db3a230677a0.png">
<img width="245" alt="Screen Shot 2022-11-15 at 11 40 13 AM" src="https://user-images.githubusercontent.com/44172267/202010749-7ec16339-7517-49b8-a5d4-b8e0e2ec8dd0.png">

malwilley reviewed (commented):
Looks like this changes some of the colors in light mode as well? not sure if that was the intention or not.

Also noticed that the syntax highlighting in the smart search bar got a little harder to read:

old: 
<img width="179" alt="CleanShot 2022-11-15 at 13 26 28@2x" src="https://user-images.githubusercontent.com/10888943/202028827-96255fc3-9c2c-4de0-b60d-fb2bfaa97a18.png">

new: 
<img width="184" alt="CleanShot 2022-11-15 at 13 26 49@2x" src="https://user-images.githubusercontent.com/10888943/202028849-d3f75581-bafd-4e3d-b98c-e59769c88f50.png">

vuluongj20 commented:
> Looks like this changes some of the colors in light mode as well? not sure if that was the intention or not.
> 
> Also noticed that the syntax highlighting in the smart search bar got a little harder to read:
> 
> old: <img alt="CleanShot 2022-11-15 at 13 26 28@2x" width="179" src="https://user-images.githubusercontent.com/10888943/202028827-96255fc3-9c2c-4de0-b60d-fb2bfaa97a18.png">
> 
> new: <img alt="CleanShot 2022-11-15 at 13 26 49@2x" width="184" src="https://user-images.githubusercontent.com/10888943/202028849-d3f75581-bafd-4e3d-b98c-e59769c88f50.png">

@malwilley 
* Light mode: Yepp the changes are intentional. The ___400 accent colors (e.g. `purple400`) are designed to be used as text colors, while the ___300 colors are for backgrounds. The changes in this PR make accent-colored text more readable in both light and dark mode. Just updated the PR description to document this.
* Search tokens: Ahh true those should be in `purple400` not `purple300`. Fixed in https://github.com/getsentry/sentry/pull/41404/commits/0347e7c481e9cf5dac54185ff7d68701911a1b4f. There are a bunch of other smaller components that still use __300 accent colors for text. I'll open a new PR to fix those!
